File: 35-user-telemetry/python-sdk/telemetry.py
import os, requests
API = "https://cost-katana-backend.store/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def get_telemetry_config():
    res = requests.get(f"{API}/user-telemetry-config",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"📊 Telemetry config: {res.json()['data']}")
    return res.json()["data"]

def update_telemetry_config(enabled, sampling):
    res = requests.put(f"{API}/user-telemetry-config",
        json={"enabled": enabled, "sampling": sampling},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print("✅ Telemetry config updated")
    return res.json()["data"]

File: 35-user-telemetry/npm-package/telemetry.ts
import axios from 'axios';
const API = 'https://api.costkatana.com/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function getTelemetryConfig() {
  const res = await axios.get(`${API}/user-telemetry-config`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('📊 Telemetry config:', res.data.data);
  return res.data.data;
}

export async function updateTelemetryConfig(enabled: boolean, sampling: number) {
  const res = await axios.put(`${API}/user-telemetry-config`, {enabled, sampling},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Telemetry config updated');
  return res.data.data;
}
